rustls-platform-verifier/src/verification/mod.rs

```diff
@@ -82,11 +82,10 @@ fn invalid_certificate(reason: impl Into<String>) -> rustls::Error {
 /// - id-kp-serverAuth
 // TODO: Chromium also allows for `OID_ANY_EKU` on Android.
 #[cfg(target_os = "windows")]
-// XXX: Windows requires that we NUL terminate EKU strings and we want to make sure that only the
-// data part of the `&str` pointer (using `.as_ptr()`), not all of its metadata.
-// This can be cleaned up when our MSRV is increased to 1.77 and C-string literals are available.
+// XXX: Windows requires that we NUL terminate EKU strings.
 // See https://github.com/rustls/rustls-platform-verifier/issues/126#issuecomment-2306232794.
-const ALLOWED_EKUS: &[*mut u8] = &["1.3.6.1.5.5.7.3.1\0".as_ptr() as *mut u8];
+const ALLOWED_EKUS: &[windows_sys::core::PCSTR] =
+    &[windows_sys::Win32::Security::Cryptography::szOID_PKIX_KP_SERVER_AUTH];
```

Comment:
Maybe? I don't really feel that strongly about it since you can still see how many are in each array and the documentation comment says what the magic string for Android is. I'll leave this up to the reviewer.